src/metpy/plots/patheffects.py

```diff
@@ -0,0 +1,245 @@
+        starts, offsets, angles = self._process_path(path, affine)
+
+        # Figure out what segments the markers should be drawn upon and how
+        # far within that segment the markers will appear.
+        segment_indices, marker_offsets = self._get_marker_locations(offsets, renderer)
+
+        # Draw the original path
+        renderer.draw_path(gc0, path, affine, rgbFace)
+
+        # Need to account for the line width in order to properly draw symbols at line edge
+        line_shift = renderer.points_to_pixels(gc.get_linewidth()) / 2
+        if self.flip:
+            line_shift = -line_shift
+
+        # Loop over all the markers to draw
+        for ind, marker_offset in zip(segment_indices, marker_offsets):
+            sym_trans = self._get_symbol_transform(renderer, marker_offset, line_shift,
+                                                   angles[ind], starts[ind])
+            renderer.draw_path(gc0, self.symbol, sym_trans, self.color)
+
+        gc0.restore()
+
+
+@exporter.export
+class ColdFront(Front):
+    """Draw a path as a cold front, with (default blue) pips/triangles along the path."""
+
+    symbol = mpath.Path([[0, 0], [1, 1], [2, 0], [0, 0]],
+                        [mpath.Path.MOVETO, mpath.Path.LINETO, mpath.Path.LINETO,
+                         mpath.Path.CLOSEPOLY])
+
+    def __init__(self, color='blue', **kwargs):
+        super().__init__(color, **kwargs)
+
+
+@exporter.export
+class WarmFront(Front):
+    """Draw a path as a warm front with (default red) scallops along the path."""
+
+    symbol = mpath.Path.wedge(0, 180).transformed(mtransforms.Affine2D().translate(1, 0))
+
+    def __init__(self, color='red', **kwargs):
+        super().__init__(color, **kwargs)
+
+
+@exporter.export
+class OccludedFront(Front):
+    """Draw an occluded front with (default purple) pips and scallops along the path."""
+
+    def __init__(self, color='purple', **kwargs):
+        self._symbol_cycle = None
+        super().__init__(color, **kwargs)
+
+    def draw_path(self, renderer, gc, path, affine, rgbFace=None):  # noqa: N803
+        """Draw the given path."""
+        self._symbol_cycle = None
+        return super().draw_path(renderer, gc, path, affine, rgbFace)
+
+    @property
+    def symbol(self):
+        """Return the proper symbol to draw; alternatives between scallop and pip/triangle."""
+        if self._symbol_cycle is None:
+            self._symbol_cycle = itertools.cycle([WarmFront.symbol, ColdFront.symbol])
+        return next(self._symbol_cycle)
+
+
+@exporter.export
+class StationaryFront(Front):
+    """Draw a stationary front as alternating cold and warm front segments."""
+
+    symbol = WarmFront.symbol
+    symbol2 = ColdFront.symbol.transformed(mtransforms.Affine2D().scale(1, -1))
+
+    def __init__(self, colors=None, **kwargs):
+        if colors is None:
+            colors = ['red', 'blue']
+        self._colors = list(map(mcolors.to_rgba, colors))
```

Comment:
`colors` is undocumented, if we want people to know how to arbitrarily choose these

Reply:
Not a bad thing to add.